### PR TITLE
Add SDK v5 stable changes

### PR DESCRIPTION
**Pull request by Sporiff**
Closes https://adjustcom.atlassian.net/browse/ADAPP-18373

### CODE REVIEW

**Comment by Sporiff** [comment]
@shashanksu @Aditi3 @uerceg The stable changes for iOS and Android are available here:

- [iOS](https://dev-docs-git-adapp-18373-sdk-v-5-full-re-0bc16d-product-content.vercel.app/en/sdk/ios?version=v5)
- [Android](https://dev-docs-git-adapp-18373-sdk-v-5-full-re-0bc16d-product-content.vercel.app/en/sdk/android?version=v5)

File: src/content/docs/sdk/android/features/privacy.mdx [hunk 1]
```diff
@@ -595,12 +595,12 @@ Adjust.trackMeasurementConsent(true);
...
 
 ```java
-public static void enableCoppaCompliance(final Context context)
+public void enableCoppaCompliance()
 ```
 
 </CodeBlock>
 
-If you need your app to be compliant with the Children's Online Privacy Protection Act (COPPA), call the `Adjust.enableCoppaCompliance` method during or after SDK intialization. This method performs the following actions:
+If you need your app to be compliant with the Children's Online Privacy Protection Act (COPPA), call the `AdjustConfig.enableCoppaCompliance` method before SDK intialization. This method performs the following actions:
```
Comment (uerceg) [inline src/content/docs/sdk/android/features/privacy.mdx:603]: Technically, `enableCoppaCompliance` is not a static method of the `AdjustConfig` class, rather an instance method (thus invoked like you stated properly in [here](https://github.com/adjust/dev-docs/pull/948/files#diff-d8df9f9de9ba9100ed3f7caab3126072c059622cd028dce5a46730512b72eea3R615), but if this is the notation you use for stating the methods in text, all good.

File: src/content/docs/sdk/android/features/privacy.mdx [hunk 2]
```diff
@@ -624,116 +624,27 @@ Adjust.enableCoppaCompliance(getApplicationContext());
...
-```java
-public static void disablePlayStoreKidsCompliance(final Context context)
-```
-
-</CodeBlock>
-
-</MinorVersion>
-
-You can unmark your app as a Play Store Kids App at any time by calling the `Adjust.disablePlayStoreKidsCompliance` method during or after SDK intialization.
+To mark your app as a Play Store Kids App, call the `AdjustConfig.enablePlayStoreKidsCompliance` method before SDK initialization.
```
Comment (uerceg) [inline src/content/docs/sdk/android/features/privacy.mdx:647]: Same as https://github.com/adjust/dev-docs/pull/948/files#r1700315603.

File: src/content/docs/sdk/android/migration/v4-to-v5.mdx [hunk 3]
```diff
@@ -630,22 +616,20 @@ adjustConfig.setCoppaCompliantEnabled(true);
...
+adjustConfig.enableCoppaCompliance()
 ```
 
 </Tab>
 <Tab title="Java" sync="java">
 
 ```java
-Adjust.enableCoppaCompliance(getApplicactionContext());
-Adjust.disableCoppaCompliance(getApplicationContext());
+AdjustConfig.enableCoppaCompliance());
```
Comment (uerceg) [inline src/content/docs/sdk/android/migration/v4-to-v5.mdx:632]: Same as [here](https://github.com/adjust/dev-docs/pull/948/files#diff-4a3cf278d9c6946d03d98cb46bf400ff9dfa1515f34361b5bf5ff8136b8c4e7dR625), better to use `adjustConfig` to indicate that it's an instance of the `AdjustConfig` class, not the class name itself.

File: src/content/docs/sdk/ios/features/privacy.mdx [hunk 4]
```diff
@@ -672,7 +672,7 @@ Adjust.trackMeasurementConsent(true)
 
 </CodeBlock>
 
-If you need your app to be compliant with the Children's Online Privacy Protection Act (COPPA), call the `[Adjust enableCoppaCompliance]` method during or after SDK intialization. This method performs the following actions:
+If you need your app to be compliant with the Children's Online Privacy Protection Act (COPPA), call the `[ADJConfig enableCoppaCompliance]` method before SDK intialization. This method performs the following actions:
```
Comment (uerceg) [inline src/content/docs/sdk/ios/features/privacy.mdx:675]: Same as https://github.com/adjust/dev-docs/pull/948/files#r1700315603, just in Obj-C world.

File: src/content/docs/sdk/ios/index.mdx [hunk 5]
```diff
@@ -24,15 +18,49 @@ Here's what you need to do before updating to SDK v5:
...
+
 // Get source directly from GitHub
-pod 'Adjust', :git => 'https://github.com/adjust/ios_sdk.git', :branch => 'v5'
+pod 'Adjust', :git => 'https://github.com/adjust/ios_sdk.git', :tag => 'v5.0.0'
+```
+
+If you're using web views in your app, add the Adjust Web Bridge by adding the following:
+
+```text
+pod 'Adjust/WebBridge', '~> 5.0.0'
```
Comment (uerceg) [inline src/content/docs/sdk/ios/index.mdx:49]:
```suggestion
pod 'Adjust/AdjustWebBridge', '~> 5.0.0'
```
/cc @Aditi3 @genadyb for sanity check.

File: src/content/docs/sdk/ios/migration/v4-to-v5.mdx [hunk 6]
```diff
@@ -1090,7 +1078,7 @@ This section covers changes to privacy features in the Adjust SDK.
 
 <MinorVersion changed="v5">
 
-In SDK v4, [COPPA compliance](/en/sdk/ios/features/privacy#coppa-compliance) is set using the `ADJConfig` class and is read during SDK initialization. This means that the value can't be updated once the SDK is initialized.
+In SDK v4, you can mark your app as COPPA compliant by calling the `[ADJConfig setCoppaCompliantEnabled]` method with a boolean value.
```
Comment (uerceg) [inline src/content/docs/sdk/ios/migration/v4-to-v5.mdx:1081]: Similar to comments above, one would not call this on the class, but on the instance of the `ADJConfig`:

```objc
[adjustConfig setCoppaComplianceEnabled:YES];
```

File: src/content/docs/sdk/android/migration/v4-to-v5.mdx (hunk 3 above)
Comment (shashanksu) [reply src/content/docs/sdk/android/migration/v4-to-v5.mdx:632]: also, there seems to be an extra `)` at the end.  Correct code:
```
adjustConfig.enableCoppaCompliance();
```

File: src/content/docs/sdk/ios/index.mdx (hunk 5 above)
Comment (Aditi3) [reply src/content/docs/sdk/ios/index.mdx:49]: Yeah, makes sense 👍🏼

File: src/content/docs/sdk/android/migration/v4-to-v5.mdx [hunk 7]
```diff
@@ -630,22 +616,20 @@ adjustConfig.setCoppaCompliantEnabled(true);
...
+adjustConfig.enableCoppaCompliance()
 ```
 
 </Tab>
 <Tab title="Java" sync="java">
 
 ```java
-Adjust.enableCoppaCompliance(getApplicactionContext());
-Adjust.disableCoppaCompliance(getApplicationContext());
+AdjustConfig.enableCoppaCompliance();
```
Comment (shashanksu) [inline src/content/docs/sdk/android/migration/v4-to-v5.mdx:632]: minor one (small `a` in adjustConfig) -
```
adjustConfig.enableCoppaCompliance();
```

**Comment by Sporiff** [comment]
@shashanksu @uerceg @Aditi3 I've merged #947 into this PR so you can review everything in one place. Once the build finishes, the content will be available in the normal structure under the following parent links:

- [Android](https://dev-docs-git-adapp-18373-sdk-v-5-full-re-0bc16d-product-content.vercel.app/en/sdk/android?version=v5)
- [iOS](https://dev-docs-git-adapp-18373-sdk-v-5-full-re-0bc16d-product-content.vercel.app/en/sdk/ios?version=v5)

**Comment by shashanksu** [comment]
@Sporiff Few comments for [Android](https://dev-docs-git-adapp-18373-sdk-v-5-full-re-0bc16d-product-content.vercel.app/en/sdk/android?version=v5):
1.  [This](https://github.com/adjust/dev-docs/pull/948#discussion_r1700393736) seems still open.
2. [Huawei Referrer API](https://dev-docs-git-adapp-18373-sdk-v-5-full-re-0bc16d-product-content.vercel.app/en/sdk/android?version=v5#huawei-referrer-api) - `-SNAPSHOT` should be removed from version.
3. [Link Resolution](https://dev-docs-git-adapp-18373-sdk-v-5-full-re-0bc16d-product-content.vercel.app/en/sdk/android/features/deep-links?version=v5#link-resolution) - `Adjust.processDeeplink` in the example is not updated.  It should be:

Kotlin -
```
AdjustLinkResolution.resolveLink(url, arrayOf("example.com"), object : AdjustLinkResolution.AdjustLinkResolutionCallback {
   override fun resolvedLinkCallback(resolvedLink: Uri) {
         val adjustDeeplink = AdjustDeeplink(resolvedLink)
         Adjust.processDeeplink(adjustDeeplink, applicationContext)
   }
})
```
Java - 
```
AdjustLinkResolution.resolveLink(url, new String[]{"example.com"}, new AdjustLinkResolution.AdjustLinkResolutionCallback() {
   @Override
   public void resolvedLinkCallback(Uri resolvedLink) {
      AdjustDeeplink adjustDeeplink = new AdjustDeeplink(resolvedLink);
      Adjust.processDeeplink(adjustDeeplink, getApplicationContext());
   }
});
```
4. Similarly for [resolve short link](https://dev-docs-git-adapp-18373-sdk-v-5-full-re-0bc16d-product-content.vercel.app/en/sdk/android/features/short-links?version=v5#set-up-the-adjust-sdk-to-resolve-short-links) chapter:

Kotlin -
```
override fun onCreate(savedInstanceState: Bundle?) {
   super.onCreate(savedInstanceState)
   setContentView(R.layout.activity_main)

   val intent = intent
   val webpageURL = intent.data
   val adjustDeeplink = AdjustDeeplink(webpageURL)
   Adjust.processAndResolveDeeplink(adjustDeeplink, this) { resolvedLink ->
      // resolvedLink contains either resolved URL (if it was unshortened)
      // or just echoed URL if it was not shortened
      Log.d("Example", resolvedLink)
   }
}
```
Java - 
```
@Override
protected void onCreate(Bundle savedInstanceState) {
   super.onCreate(savedInstanceState);
   setContentView(R.layout.activity_main);

   Intent intent = getIntent();
   Uri webpageURL = intent.getData();
   AdjustDeeplink adjustDeeplink = new AdjustDeeplink(webpageURL);
   Adjust.processAndResolveDeeplink(adjustDeeplink, this, new   OnDeeplinkResolvedListener() {
      @Override
      public void onDeeplinkResolved(String resolvedLink) {
            // resolvedLink contains either resolved URL (if it was unshortened)
            // or just echoed URL if it was not shortened
            Log.d("Example", resolvedLink);
      }
   });
}
```
(same applies for `onNetIntent` method example)

**Comment by Aditi3** [comment]
Hi @Sporiff 

- The following lines can be removed and not required anymore -

`You can also use the included WebViewJavascriptBridge by setting the bridgeRegister property of your AdjustBridge instance. See the [library’s documentation](https://github.com/marcuswestin/WebViewJavascriptBridge#usage) for usage information.`

- I think better would be point to GitHub URL: Replace `pod 'Adjust/AdjustWebBridge', '~> 5.0.0'` with `pod 'Adjust/AdjustWebBridge', :git => 'https://github.com/adjust/ios_sdk.git', :tag => 'v5.0.0'`

File: src/content/docs/sdk/android/features/short-links.mdx [hunk 8]
```diff
@@ -85,9 +87,12 @@ protected void onCreate(Bundle savedInstanceState) {
 ```kotlin
 override fun onNewIntent(intent: Intent?) {
    super.onNewIntent(intent)
+   setContentView(R.layout.activity_main)
```
Comment (shashanksu) [inline src/content/docs/sdk/android/features/short-links.mdx:90]: This line is redundant and should be removed.  Only applicable for `onCreate` method.

File: src/content/docs/sdk/android/features/short-links.mdx [hunk 9]
```diff
@@ -85,9 +87,12 @@ protected void onCreate(Bundle savedInstanceState) {
...
 
+   val intent = intent
```
Comment (shashanksu) [inline src/content/docs/sdk/android/features/short-links.mdx:92]: This line is redundant and should be removed. Only applicable for onCreate method.

File: src/content/docs/sdk/android/features/short-links.mdx [hunk 10]
```diff
@@ -102,15 +107,18 @@ override fun onNewIntent(intent: Intent?) {
 @Override
 protected void onNewIntent(Intent intent) {
    super.onNewIntent(intent);
+   setContentView(R.layout.activity_main);
```
Comment (shashanksu) [inline src/content/docs/sdk/android/features/short-links.mdx:110]: This line is redundant and should be removed. Only applicable for onCreate method.

File: src/content/docs/sdk/android/features/short-links.mdx [hunk 11]
```diff
@@ -102,15 +107,18 @@ override fun onNewIntent(intent: Intent?) {
...
 
+   Intent intent = getIntent();
```
Comment (shashanksu) [inline src/content/docs/sdk/android/features/short-links.mdx:112]: This line is redundant and should be removed. Only applicable for onCreate method.

**Review by shashanksu — APPROVED** [review]
Thank you @Sporiff for the speedy fixes.

**Review by uerceg — APPROVED** [review]
:shipit:

**Pull request merged**